app/src/controllers/reservation/services/ReservationService.ts

```diff
@@ -52,6 +57,47 @@ export const getNextAvailableDate = (reservationDate: Date, menuDuration: number
   return nextAvailableDate
 }
 
+export const checkInsideShopSchedule = (shopSchedule: {startTime?: string,
+  endTime?: string}, reservationDate: Date, menuDuration: number) => {
+  if (shopSchedule.startTime !== undefined
+    && shopSchedule.endTime !== undefined) {
+    const rEndTime = getNextAvailableDate(reservationDate, menuDuration)
+    const sStartTime = new Date(
+      `${reservationDate.getFullYear()}-${reservationDate.getMonth() + 1}-${reservationDate.getDate()}
+      ${shopSchedule.startTime}`,
+    )
+    const sEndTime = new Date(
+      `${reservationDate.getFullYear()}-${reservationDate.getMonth() + 1}-${reservationDate.getDate()}
+      ${shopSchedule.endTime}`,
+    )
```

Comment:
ここちょい力付きで行きました。

Reply:
maybe using
```typescript
const convertToUnixTime = (time:string): number => new Date(`January 1, 2020 ${time}`).getTime()
```
found inside the shop service of shop controller would have been easier?

Reply:
about that i did that at first but i need the dates to be the same as the reservation dates too because then we use getTime() apparently it uses the dates as well